### PR TITLE
docs/release: generate the changelog as rst instead of md

### PR DESCRIPTION
**Pull request by beardypig**
I took a stab at modifying the release script so that the change log is generated as rst as discussed in #238 and #271. 

@cdrage should take a look at this.

### CODE REVIEW

**Comment by gravyboat** [comment]
With this PR I believe we're good to merge: https://github.com/streamlink/streamlink/pull/328, any concerns? If not I'll merge that one in, then we can get this one merged once others have reviewed it.

**Comment by beardypig** [comment]
I think #328 conflicts with the release script a bit, the release script creates a github Release page as well. I think just the part where it add the built installer from travis to the github tag is sufficient for now.

**Comment by beardypig** [comment]
I think this portion of the `.travis.yml` c42f96d77437a2f46e0279bae75f3571c825b06c will be enough for now, just to get the Windows installer automatically uploaded.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-streamlink 0.2.0 (12-16-2016)
+streamlink 0.2.0 (2106-12-16)
```
Comment (bastimeyer) [inline CHANGELOG.rst]: 2016

Comment (beardypig) [inline CHANGELOG.rst]: Damn that time capsule

**Comment by gravyboat** [comment]
@bastimeyer That would be great, then we can get both of these merged in and be up to date across the board.

**Comment by beardypig** [comment]
@gravyboat @bastimeyer we'll have to make sure @cdrage is up-to-date (as he's the de-facto release manager) with the release process if the release page is created automatically, as the current release script makes the release page - maybe we need to update that script at the same time. We could do the Windows installer upload and the other changes separately, so we get the advantages of the automatic Windows builds but don't have to update the release script. 

@bastimeyer it would be good to update the release script to support rst :)

**Comment by cdrage** [comment]
This looks good to me :) Next release I can test this out and see if it generates it correctly! But good job :+1:

**Comment by gravyboat** [comment]
Closing as requested here: https://github.com/streamlink/streamlink/pull/328#issuecomment-272094312

**Pull request closed without merging**